actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ShadowUserMigrationScheduler.java

```diff
@@ -0,0 +1,312 @@
+        }
+        return result;
+    }
+
+    /**
+     * this method will prepare the shawdow user object and insert the record into shawdow_user table
+     * @param createdBy
+     * @param processId
+     * @param migrationUser
+     */
+    private void insertShadowUserToDb(String createdBy,String processId, MigrationUser migrationUser) {
+        ShadowUser shadowUser = new ShadowUser.ShadowUserBuilder()
+                .setUserExtId(migrationUser.getUserExternalId())
+                .setOrgExtId(migrationUser.getOrgExternalId())
+                .setEmail(migrationUser.getEmail())
+                .setPhone(migrationUser.getPhone())
+                .setUserId(null)
+                .setAddedBy(createdBy)
+                .setUserIds(null)
+                .setChannel(migrationUser.getChannel())
+                .setName(migrationUser.getName())
+                .setProcessId(processId)
+                .setClaimedOn(null)
+                .setClaimStatus(ClaimStatus.UNCLAIMED.getValue())
+                .setUserStatus(getInputStatus(migrationUser.getInputStatus()))
+                .setUpdatedOn(null)
+                .build();
+        Map<String, Object> dbMap = mapper.convertValue(shadowUser, Map.class);
+        dbMap.put(JsonKey.CREATED_ON, new Timestamp(System.currentTimeMillis()));
+        if(!isOrgExternalIdValid(migrationUser)) {
+            dbMap.put(JsonKey.CLAIM_STATUS,ClaimStatus.REJECTED.getValue());
+        }
+        Response response = cassandraOperation.insertRecord(JsonKey.SUNBIRD, JsonKey.SHADOW_USER, dbMap);
+        ProjectLogger.log("ShadowUserMigrationScheduler:insertShadowUser: record status in cassandra ".concat(response.getResult() + ""), LoggerEnum.INFO.name());
+
+    }
+
+    private int getInputStatus(String inputStatus) {
+        if (inputStatus.equalsIgnoreCase(JsonKey.ACTIVE)) {
+            return ProjectUtil.Status.ACTIVE.getValue();
+        }
+        return ProjectUtil.Status.INACTIVE.getValue();
+    }
+
+    private void updateUser(MigrationUser migrationUser, ShadowUser shadowUser) {
+        updateUserInShadowDb(migrationUser,shadowUser);
+    }
+
+    private void updateUserInShadowDb(MigrationUser migrationUser, ShadowUser shadowUser) {
+        if(!isSame(shadowUser,migrationUser)){
+            Map<String, Object> propertiesMap = new HashMap<>();
+            propertiesMap.put(JsonKey.EMAIL, migrationUser.getEmail());
+            propertiesMap.put(JsonKey.PHONE, migrationUser.getPhone());
+            propertiesMap.put(JsonKey.NAME, migrationUser.getName());
+            propertiesMap.put(JsonKey.ORG_EXT_ID, migrationUser.getOrgExternalId());
+            propertiesMap.put(JsonKey.UPDATED_ON, new Timestamp(System.currentTimeMillis()));
+            propertiesMap.put(JsonKey.USER_STATUS,getInputStatus(migrationUser.getInputStatus()));
+            if(!isOrgExternalIdValid(migrationUser)) {
+                propertiesMap.put(JsonKey.CLAIM_STATUS, ClaimStatus.REJECTED.getValue());
```

Comment:
Check condition for rejected (in PRD)

Reply:
@manzarul  this rejected status should be failed right?

Reply:
@manzarul  sir I have added `EXTERNALIDMISMATCH` as ENUM which will be UPDATE IF USER FAILS THIS CHECK OF ORGEXTERNALID